src/test/java/ru/ayaz/UserMessageHandlerTest.java

```diff
@@ -0,0 +1,103 @@
+package ru.ayaz;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.net.Socket;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.Mockito.*;
+
+public class UserMessageHandlerTest {
+
+    private UserMessageDistributor messageDistributor;
+
+    @Before
+    public final void before() {
+        messageDistributor = spy(new UserMessageDistributor());
+    }
+
+    @Test
+    public void shouldBeAbleToEnqueueMessage() throws InterruptedException {
+        UserMessage someMessage = new UserMessage("Ayaz", "Hello");
+
+        messageDistributor.enqueueMessage(someMessage);
+
+        assertTrue(messageDistributor.queueContainsMessage(someMessage));
+    }
+
+    @Test
+    public void shouldBeAbleToGetEnqueuedMessage() throws InterruptedException {
+        UserMessage someMessage = new UserMessage("Ayaz", "Hello");
+        messageDistributor.enqueueMessage(someMessage);
+
+        UserMessage gottenMessage = messageDistributor.takeMessage();
+
+        assertEquals(someMessage, gottenMessage);
+    }
+
+
+    @Test
+    public void shouldWarnAboutUnknownCommandWhenMalformedCommandTaken() throws Exception {
+        Socket socket = mock(Socket.class);
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream(100);
+        when(socket.getOutputStream()).thenReturn(outputStream);
+        when(socket.getInputStream()).thenReturn(new ByteArrayInputStream(new byte[1]));
```

Comment:
&#x1F34E; **Issue**
You don't need to create real ByteArrayInputStream here. Mock will be just fine.
In general, if you can mock it - do so. Why ? I sent an article for you to read in our group chat about TDD.
This comment concerns not only this line, but every other similar case in your project, please, do so for every test. (In this test there are other things you should mock besides ByteArrayInputStream - try to figure out this one on your own ;) )